course_discovery/apps/edx_elasticsearch_dsl_extensions/management/commands/search_index.py

```diff
@@ -0,0 +1,246 @@
+import contextlib
```

Comment:
This command is called `search_index`, but is replacing `update_index`, yeah? Might help to keep the same name.

Reply:
`search_index` is the native command for `django-elasticsearch-dsl` https://django-elasticsearch-dsl.readthedocs.io/en/latest/management.html. 
I thought it would be wise to just extend it and add the `--update` flag, under which the sanity check feature will be implemented. Those if we call `search_index --update`, the indexes will be created as before.
But since you paid attention to this, it makes sense to keep the old functionality that users are used to and create the `update_index` command and do it as it was before. Okay, that makes sense. I moved the logic and now everything as before `$ ./manage.py update_index --disable-change-limit` works for us too. Here is this commit https://github.com/raccoongang/course-discovery/pull/12/commits/aa624847e8aa0031977e520f306e601487478d5c.

Note that `update_index` is the native `django-haystack` command.

Reply:
But also want to notice that a user can use the commands from https://django-elasticsearch-dsl.readthedocs.io/en/latest/management.html. This is also possible now. There will be no sanity check feature, perhaps this option will be of interest to users. They now have a choice of how to work with their indexes. I have also overridden `search_index --update` so that indexes are created with aliases and timestamped.